ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -171,17 +146,70 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
 static int
 S3FileSystemTerminate(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session)
 {
-    S3_FILE_SYSTEM *fs;
+    S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
 
     UNUSED(session); /* unused */
 
-    fs = (S3_FILE_SYSTEM *)fileSystem;
     delete (fs->conn);
     free(fs);
 
     return (0);
 }
 
+/*
+ * S3DirectoryList --
+ *     Return a list of object names for the given location.
+ */
+static int
+S3DirectoryList(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *directory,
+  const char *prefix, char ***dirlistp, uint32_t *countp)
+{
+    S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
+    std::vector<std::string> objects;
+    int ret;
+    if (ret =
+          fs->conn->ListObjects(std::string(directory), std::string(prefix), objects, *countp) != 0)
```

Comment:
Do we need to pass an argument of ```countp``` to the ListObjects. Is it not the same count as ```objects.size()```